telemetry/csharp/AwsToolkit.Telemetry.Events/Core/MetricDatumExtensionMethods.cs

```diff
@@ -57,5 +61,27 @@ public static void AddMetadata(this MetricDatum metricDatum, string key, int val
         {
             metricDatum.AddMetadata(key, value.ToString(System.Globalization.CultureInfo.InvariantCulture));
         }
+
+        /// <summary>
+        /// If the transform function isn't null, invoke it and assign metric datum to it's result
+        /// </summary>
+        public static MetricDatum InvokeTransform(this MetricDatum metricDatum, ILog logger,
```

Comment:
We don't need to pass in the logger. This class can just use its own.
Its okay if we don't test that something was logged. The most interesting test to have is that an exception does not escape form calling `InvokeTransform` (which would break the call stack and possibly crash the IDE)